pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/MessageDeduplication.java

```diff
@@ -493,6 +501,12 @@ public synchronized void purgeInactiveProducers() {
         long minimumActiveTimestamp = System.currentTimeMillis() - TimeUnit.MINUTES
                 .toMillis(pulsar.getConfiguration().getBrokerDeduplicationProducerInactivityTimeoutMinutes());
 
+        // if not enabled just clear all inactive producer record.
+        if (!isEnabled()) {
```

Comment:
Why put the check here?

Reply:
if we enable and disable deduplication for master code logic. the inactive producer will be delay for getBrokerDeduplicationProducerInactivityTimeoutMinutes to clean up.

Do we want to check in Topic.checkMessageDeduplicationInfo to check if deduplication is enabled if not just skip call purgeInactiveProducers ? If yes the inactive producer should be clear in disable deduplication logic.

Reply:
Hi @lifepuzzlefun 
I think we'd better remove the check here, because we add the check in `producerRemoved` and `producerAdded` .